packages/typespec-python/test/generated/azure-core-lro-rpc-legacy/_specs_/azure/core/lro/rpc/legacy/_operations/_operations.py

```diff
@@ -66,7 +66,7 @@ def build_legacy_create_job_request(**kwargs: Any) -> HttpRequest:
 
 
 class LegacyClientOperationsMixin(LegacyClientMixinABC):
-    def _create_job_initial(self, body: Union[_models.JobData, JSON, IO], **kwargs: Any) -> Optional[_models.JobResult]:
+    def _create_job_initial(self, body: Union[_models.JobData, JSON, IO], **kwargs: Any) -> _models.JobResult:
```

Comment:
@tadelesh or @msyyc can you guys double check why this is not returning optional anymore? i thought it wouldn't matter but we're now failing tests. ideally want to release tomorrow since a service team is blocked. Thanks!

Reply:
In previous logic of lro, we just set the body to the lro logic result when it has body, it means we will not set body for any init response of 202,201 etc. So, this issue is introduced when you want to set all response body to lro logic result. I don't think it is right. And I check the tsp definition of lro from healthinsights, they are using [TResponse as 202 response](https://github.com/Azure/azure-rest-api-specs/blob/1c36936030d8a7266b1162293615b3eaa1c57b3f/specification/cognitiveservices/HealthInsights/healthinsights.common/primitives.tsp#L16C59-L16C59), not defined 200 response for lro. I don't know much about their service's behavior, but I doubt if it is a right tsp definition for lro.